src/cursor/cur_std.c

```diff
@@ -1265,30 +1276,16 @@ __wt_cursor_bound(WT_CURSOR *cursor, const char *config)
             WT_ERR(__wt_buf_set(session, &cursor->lower_bound, key.data, key.size));
         } else
             WT_ERR_MSG(session, EINVAL,
-              "setting bounds only accepts \"upper\" or \"lower\" as the configuration");
-
-    } else if (WT_STRING_MATCH("clear", cval.str, cval.len)) {
-        /* Inclusive should not be supplied from the application with action clear configuration. */
-        if (__wt_config_getones(session, config, "inclusive", &cval) != WT_NOTFOUND)
-            WT_ERR_MSG(session, EINVAL,
-              "clearing bounds is not compatible with the inclusive configuration");
-
-        /*
-         * Check if there is a lower or upper specified bound config. If there are no specified
-         * bounds, both the upper and lower bound will be cleared.
-         */
-        WT_ERR(__wt_config_gets(session, cfg, "bound", &cval));
-        if (cval.len == 0 || WT_STRING_MATCH("upper", cval.str, cval.len)) {
-            F_CLR(cursor, WT_CURSTD_BOUND_UPPER | WT_CURSTD_BOUND_UPPER_INCLUSIVE);
-            __wt_buf_free(session, &cursor->upper_bound);
-            WT_CLEAR(cursor->upper_bound);
-        }
-        if (cval.len == 0 || WT_STRING_MATCH("lower", cval.str, cval.len)) {
-            F_CLR(cursor, WT_CURSTD_BOUND_LOWER | WT_CURSTD_BOUND_LOWER_INCLUSIVE);
-            __wt_buf_free(session, &cursor->lower_bound);
-            WT_CLEAR(cursor->lower_bound);
-        }
-    }
+              "a bound must be specified when setting bounds, either \"lower\" or \"upper\"");
+    } else if ((cfg2 = strstr(cfg1, "clear")) != NULL) {
+        F_CLR(cursor, WT_CURSTD_BOUND_ALL);
+        __wt_buf_free(session, &cursor->upper_bound);
+        __wt_buf_free(session, &cursor->lower_bound);
+        WT_CLEAR(cursor->upper_bound);
```

Comment:
Do we need the WT_CLEAR though we've already called the __wt_buf_free that does a memset to 0?